dev/io.openliberty.microprofile.openapi.ui.internal_fat/fat/src/io/openliberty/microprofile/openapi/ui/internal/fat/tests/UIOauthTest.java

```diff
@@ -0,0 +1,226 @@
+    }
+
+    @After
+    public void teardownTest() throws Exception {
+        //close the browser before stopping the server to reduce changes of open connections
+        driver.quit();
+        server.stopServer();
+    }
+
+    @Test
+    public void defaultPathOAuthTest() throws Exception {
+        // Set the OAuth Endpoint configuration
+        server.addEnvVar(UI_PATH_PROPERTY,UI_PATH_VALUE);
+
+        server.startServer();
+
+        OAuthTest(UI_PATH_VALUE);
+
+    }
+
+    @Test
+    public void customPathOAuthTest() throws Exception{
+        // Set the mpOpenAPI endpoint configuration
+        ServerConfiguration config = server.getServerConfiguration();
+        config.getMpOpenAPIElement().setUiPath(CUSTOM_UI_PATH_VALUE);
+        config.getMpOpenAPIElement().setDocPath(DOC_PATH_VALUE);
+        server.updateServerConfiguration(config);
+        // Set the OAuth Endpoint configuration
+        server.addEnvVar(UI_PATH_PROPERTY, CUSTOM_UI_PATH_VALUE);
+
+        server.startServer();
+
+        OAuthTest(CUSTOM_UI_PATH_VALUE);
+    }
+
+
+    public void OAuthTest(String uiPath){
+        //Need to use HTTPS path for the test for OAuth to actually work
+        driver.get("https://"+HOST+":" + server.getHttpDefaultSecurePort() + uiPath);
+
+        //Store the id of the original window as we need to know when we go through the oauth login flow
+        //We don't need to wait for the page to load to know which one we are after
+        String originalWindow = driver.getWindowHandle();
+
+        // Check the title loads as initial validation page has loaded
+        WebElement title = waitForElement(driver, By.cssSelector("h2.title"), LONG_WAIT);
+        assertThat("Page title", title.getText(), Matchers.containsString("Generated API"));
+
+        //check that the operation is "unlocked"
+        //we will use the lock object later to check that it has been updated
+        WebElement lock = waitForElement(driver, By.cssSelector("button.authorization__btn.unlocked"));
+
+        //Check the Authorize button is available and click it
+        WebElement authorizeButton = waitForElement(driver, By.xpath("//span[contains(.,'Authorize')]"));
+        //assertThat("Authorize button", authorizeButton.findElement(By.cssSelector("span")).getText(), Matchers.containsString("Authorize"));
+        authorizeButton.click();
+
+        //check that the modal loads
+        WebElement oauthModal = waitForElement(driver, By.cssSelector("div.modal-ux-header"));
+        WebElement oauthModalHeader = oauthModal.findElement(By.cssSelector("h3"));
+        assertThat("", oauthModalHeader.getText(),Matchers.containsString("Available authorizations"));
+
+        // Get the OAuth Authentication container
+        //WebElement authContainer = oauthModal.findElement(By.cssSelector("div.auth-container"));
+        WebElement authContainer = waitForElement(driver, By.cssSelector("div.auth-container"));
```

Comment:
More commented out code. Was the `auth-container` not found under the modal or did it take a moment to appear? You should be able to pass `oauthModel` to `waitForElement` if the latter case.